pages/api/attributes/[attributeId].ts

```diff
@@ -0,0 +1,69 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import {
+  getAttribute,
+  updateAttribute,
+  deleteAttribute,
+} from '../../../server/actions/Attributes'
+import { ApiError, Attribute } from '../../../utils/types'
+import { unstable_getServerSession } from 'next-auth'
+import { authOptions } from '../auth/[...nextauth]'
+
+// @route GET api/attributes/[attributeId] - Returns a single Attribute object given by a attributeId - Private
+// @route PUT api/attributes/[attributeId] - Updates an existing Attribute object (identified by categoryId) with a new Category object - Private
+// @route DELETE api/attributes/[attributeId] - Deletes a single Atttribute object (identified by attributeId) - Private
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    // ensure user is logged in
+    const session = await unstable_getServerSession(req, res, authOptions)
+    if (!session) {
+      throw new ApiError(401, 'You must be authenticated to make this request.')
+    }
```

Comment:
This same logic is also in `/pages/api/attributes/index.ts`, and twice in PR #14. I'm thinking it might make sense to make a function for this in `/utils/auth.ts`. I'd recommend renaming the current one to "userEndpointServerAuth" and calling this new one "serverAuth"

Could you either implement this function or make a story for it?

Reply:
Implemented with f2a06aa and ae2ec63!